src/modules/infobox/components/FeatureElevationProfilePlot.vue

```diff
@@ -105,238 +122,304 @@ export default {
+         * Same things as the minimum value, will give a 10% of the elevation delta as buffer, with
+         * a minimum of 5meters
+         */
+        yAxisMaxValue() {
+            return (
+                Math.ceil(this.elevationProfile.maxElevation) +
+                Math.max(Math.ceil(this.tenPercentOfElevationDelta), 5)
+            )
+        },
+        /** Definition of the data ChartJS will show, with some styling configuration too */
+        chartJsData() {
+            return {
+                datasets: [
+                    {
+                        label: `${this.$t('elevation')}`,
+                        data: this.elevationProfile.points,
+                        parsing: {
+                            xAxisKey: 'dist',
+                            yAxisKey: 'elevation',
+                        },
+                        pointRadius: 1,
+                        pointHoverRadius: 3,
+                        borderColor: 'rgb(255, 99, 132)',
+                        borderWidth: 1,
+                        fill: {
+                            target: 'origin',
+                            above: 'rgba(255, 99, 132, 0.7)',
+                        },
+                        // smooth up a bit the line (can be removed/reverted to 'default' if we want a sharper line)
+                        cubicInterpolationMode: 'monotone',
+                    },
+                ],
+            }
+        },
+        /** Definition of axis for the profile chart */
+        chartJsScalesConfiguration() {
+            return {
+                x: {
+                    type: 'linear',
+                    max: this.elevationProfile.maxDist,
+                    title: {
+                        display: true,
+                        text: `${this.$t('distance_label')} [${this.unitUsedOnDistanceAxis}]`,
+                        font: {
+                            weight: 'bold',
+                        },
+                        // removing the padding so that we gain a bit of vertical space
+                        padding: 0,
+                    },
+                    ticks: {
+                        // processing distance number to be more human-readable
+                        callback: (val) => round(val * this.factorToUseForDisplayedDistances, 1),
+                    },
+                },
+                y: {
+                    title: {
+                        display: true,
+                        text: `${this.$t('elevation')} [m]`,
+                        font: {
+                            weight: 'bold',
+                        },
+                    },
+                    min: this.yAxisMinimumValue,
+                    max: this.yAxisMaxValue,
+                },
+            }
+        },
+        /** Defines how the mouse-over tooltip will behave */
+        chartJsTooltipConfiguration() {
+            return {
+                enabled: false,
+                external: ({ chart, tooltip }) => {
```

Comment:
I would avoid returning complex method/callback in computed properties which are cached and reactive.

Reply:
then maybe move that to some methods, and have watchers to update the ChartJS instances in case of update... I'll look into it, that's true it's not optimal to have such big objects in the reactive loop